### DISCUSSION TITLE
Modifier for children

### DISCUSSION THREAD

**Discussion by eliasu** [statamic/cms#9810] [Q&A]
Hi there, 

I'm trying to write a modifier, which gives back a boolean, if the provided page has children (in an structured collection) or not.

This is my html structure:

```antlers
{{ collection:pages as="all_pages" sort="order" }}
    {{ all_pages group_by="parent:title" }}
        {{ groups }}
            <ul>
                {{ items scope="item" }}
                    <li>
                        <a href="{{ item.parent:url }}">
                            {{ item.parent:title }}
                        </a>
                        {{ if item.id | haschild }}
                            <div>Show me if there are children</div>
                        {{ /if }}
                    </li>
                {{ /items }}
            </ul>
        {{ /groups }}
    {{ /all_pages }}
{{ /collection:pages }}
```

As you can see I'm basically 
1. looping over every page
2. group the pages by their parents title.
3. create a link with the items tile and href
4. try to find out, if the page has children with the "haschild" modifier. 

Here is the modifier, i tried a lot, but I give you a simple example which does not work. And I don't understand why:

```php
class Haschild extends Modifier
{
    public function index($value, $params, $context)
    {
        $entry = \Statamic\Facades\Entry::find($value);
        $page = $entry->page();
        $children = $page->children();

        debugbar()->info($children);
    }
}
```
Unfortunately this does not work:
`[03:21:54] LOG.notice: Error in [haschild] modifier: Call to undefined method Statamic\Structures\Page::children()`

If I debug `$page` i will get something like this:
![Bildschirmfoto 2024-04-03 um 03 21 41](https://github.com/statamic/cms/assets/11287110/b64dae64-e241-47fc-ae7d-25d57045d830)

So **somewhere**, there is this children array. But how to access it properly? 

Thanks for your help!

**Comment by jasonvarga** [comment]
`$page->pages()`!

**Reply by eliasu** [reply]
Thanks for you answer Jason, 
but … I don't get it. Where should I add this snippet? 

```php
<?php

namespace App\Modifiers;

use Statamic\Modifiers\Modifier;

class Haschild extends Modifier
{

    public function index($value, $params, $context)
    {
        $entry = \Statamic\Facades\Entry::find($value);
        $page = $entry->page();
        $children = $page->pages();
        
        debugbar()->info($children);

        if ($children != null) {
            return true;
        }
        return false;
    }
}

```
